Add Pokemon lookup

rwnx opened the pull request:
* I've added a lookup function that also validates pokedex number. It's
  also had the side effect of adding some validation to the input args.
* added a screenshot and some additional usage information

Commit:
Add Pokemon lookup

* I've added a lookup function that also validates pokedex number. It's
  also had the side effect of adding some validation to the input args.
* added a screenshot and some additional usage information

danjordan commented on index.js:
```diff
@@ -4,12 +4,25 @@ const ora = require("ora");
+    const pokemon2 = findPokedexNumber(args[1]);
 
-    const url = ["https://pokemon.alexonsager.net", ...args].join("/");
+    if (!pokemon1) {
```
https://pokemon.alexonsager.net returns Missingno for unknown Pokemon.

We can remove these two Errors to keep that functionality.

jerometwell replied:
Sure thing! I wasn't sure whether we wanted to keep this behaviour. I'll remove it :)

danjordan commented on lookup.js:
```diff
@@ -0,0 +1,182 @@
+ * returns falsey/null when input invalid
+ * @param {String|Number} pokemonInput
+ */
+function findPokedexNumber(pokemonInput) {
```
Because https://pokemon.alexonsager.net gives us missingno I think we can remove the validation from here.

Something like this...

`function findPokedexNumber(pokemon = "") {
  const index = pokedex.findIndex(
    (x) => x.name.toLowerCase() === pokemon.toLowerCase()
  );

  if (index >= 0) return index + 1;

  return pokemon;
}`

jerometwell replied:
won't this remove the original behaviour, which was to specify number? Or did you want to keep that?

danjordan replied:
Shouldn't remove it. If it can't find an index it returns the original input.

jerometwell replied:
Ah, I see now. Do you think there's value in forcing a missingno when the user specifies a bad name e.g. 
```
findPokedexNumber("notarealpokemon"); 
```
Or do you think it's better to just pass that directly to the api ?

danjordan replied:
I think it's better to forward it to the API

danjordan commented on README.md:
````diff
@@ -3,9 +3,17 @@
+Pokemon can be specified as names or pokedex numbers:
+```
+pokefusion charizard bulbasaur
+pokefusion 6 1
````
Can we pop an example in here for Nidoran F/M or Farfetch'd so people know how to handle Pokemon with spaces or non alpha chars.

Commit:
Add examples for complex names/name behaviour

Commit:
Remove input validation, passthru to api

* delegate the valdiation concerns down to the called api. You should be
able to trigger a merge with missingno by putting an out of range number
or an unknown pokemon as an argument.
